lib/expand/join.go

```diff
@@ -142,6 +146,7 @@ func (p *Peer) Run(listener net.Listener) error {
 			p.WithError(err2).Warn("Failed to mark operation as failed.")
 		}
 	}
+	p.WithField("exit-error", err).Info("Exit with error.")
```

Comment:
Error or Warn?

Reply:
It is more informational than it is an error or warning. I might even remove it as debugging aid.